src/libraries/System.Numerics.Tensors/tests/TensorPrimitivesTests.cs

```diff
@@ -702,5 +702,123 @@ public static void TanhTensor_ThrowsForTooShortDestination(int tensorLength)
 
             AssertExtensions.Throws<ArgumentException>("destination", () => TensorPrimitives.Tanh(x, destination));
         }
+
+        [Theory]
+        [MemberData(nameof(TensorLengths))]
+        public static void CosineSimilarity_ThrowsForMismatchedLengths_x_y(int tensorLength)
+        {
+            float[] x = CreateAndFillTensor(tensorLength);
+            float[] y = CreateAndFillTensor(tensorLength - 1);
+
+            Assert.Throws<ArgumentException>(() => TensorPrimitives.CosineSimilarity(x, y));
+        }
+
+        [Theory]
+        [InlineData(new float[] { 3, 2, 0, 5 }, new float[] { 1, 0, 0, 0 }, 0.49f)]
+        [InlineData(new float[] { 1, 1, 1, 1, 1, 0 }, new float[] { 1, 1, 1, 1, 0, 1 }, 0.80f)]
```

Comment:
These tests will be insufficient for vectorized implementations as they're not large enough to trigger paths for some vector widths. It'd also be good to validate empty inputs.

Reply:
Yeah, I know they will be insufficient for the vectorized tests, but since none of these are yet I wasn't too worried. Would you like me to fix that now though or when we get the vectorized implementations in? Happy either way.

Reply:
Let's get it merged and then we can add more.